### PR TITLE
chore(tooling): Use wasmer instead of wasmi

### LINKED ISSUE #3214: Sync the version of wasmi
### Problem to Solve

we have different versions of wasmi used in our workspace for now

workspace: https://github.com/gear-tech/gear/blob/master/Cargo.toml#L165
sandbox : https://github.com/gear-tech/gear/blob/master/sandbox/host/Cargo.toml#L21
wasm-builder( this uses wasmi as dev-dep ): https://github.com/gear-tech/gear/blob/master/utils/wasm-builder/Cargo.toml#L33

### Possible Solution

spread the wasmi of sandbox-host to the workspace

### Notes

_No response_

### PR DESCRIPTION
**Pull request by ark0f**
Resolves #3214

### CODE REVIEW

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
Seems that we need to publish our fork. So we could always depend on crates btw @ark0f @clearloop

**Review by ByteNacked — APPROVED** [review]

File: sandbox/host/Cargo.toml [hunk 1]
```diff
@@ -20,9 +20,9 @@ defer.workspace = true
 environmental.workspace = true
 thiserror.workspace = true
 log = { workspace = true, features = ["std"] }
-sandbox-wasmer.workspace = true
-sandbox-wasmer-types.workspace = true
-sandbox-wasmi = { workspace = true, features = ["virtual_memory"] }
+wasmer = { workspace = true, features = ["singlepass"] }
+wasmer-types.workspace = true
+sandbox-wasmi.workspace = true
```
Comment (clearloop) [inline sandbox/host/Cargo.toml:25]: Yes, `table_like` makes more sense for this kind of cases !

File: Cargo.toml [hunk 2]
```diff
@@ -303,11 +302,7 @@ ethexe-validator = { path = "ethexe/validator", default-features = false }
 ethexe-rpc = { path = "ethexe/rpc", default-features = false }
 ethexe-common = { path = "ethexe/common" }
 
-# Common executors between `sandbox-host` and `calc-stack-height`
-sandbox-wasmer = { package = "wasmer", version = "4.3.4", features = [
-    "singlepass",
-] }
-sandbox-wasmer-types = { package = "wasmer-types", version = "4.3.4" }
+# Common executor between `sandbox-host` and `calc-stack-height`
 sandbox-wasmi = { package = "wasmi", git = "https://github.com/gear-tech/wasmi", branch = "v0.13.2-sign-ext", features = [
```
Comment (clearloop) [inline Cargo.toml:306]: I think we'd better gather all `wasmi` dependencies in the global config for easy debugging like in this `sandbox-wasmi` form, except this case there is also a `gwasmi` used in `sandbox-host` which is our forked version

**Review by clearloop — APPROVED** [review]
the version management of wasmi is really a nightmare, could not agree more with using wasmer/wasmtime instead of `wasmi` in no wasm required environments for reducing the workload on version managements !

btw we have the forked version wasmi `gwasmi` 😅  cc @breathx

File: Cargo.toml [hunk 3]
```diff
@@ -303,11 +302,7 @@ ethexe-validator = { path = "ethexe/validator", default-features = false }
 ethexe-rpc = { path = "ethexe/rpc", default-features = false }
 ethexe-common = { path = "ethexe/common" }
 
-# Common executors between `sandbox-host` and `calc-stack-height`
-sandbox-wasmer = { package = "wasmer", version = "4.3.4", features = [
-    "singlepass",
-] }
-sandbox-wasmer-types = { package = "wasmer-types", version = "4.3.4" }
+# Common executor between `sandbox-host` and `calc-stack-height`
```
Comment (StackOverflowExcept1on) [inline Cargo.toml:305]: `calc-stack-height` uses `wasmer`
```suggestion
# Common executor between `sandbox-host`
```

File: gcli/src/result.rs [hunk 4]
```diff
@@ -48,7 +48,7 @@ pub enum Error {
     #[error("Wasm execution error {0}")]
     WasmExecution(String),
     #[error("Wasmi execution error {0}")]
-    Wasmi(wasmi::Error),
+    Wasmi(wasmer::RuntimeError),
```
Comment (StackOverflowExcept1on) [inline gcli/src/result.rs:51]:
```suggestion
    Wasmer(wasmer::RuntimeError),
```

**Review by StackOverflowExcept1on — APPROVED** [review]

File: gcli/src/meta/executor.rs [hunk 5]
```diff
@@ -63,33 +63,27 @@ fn execute(wasm: &[u8], method: &str) -> Result<Vec<u8>> {
...
-        .get_export(&store, method)
-        .and_then(Extern::into_func)
-        .ok_or_else(|| anyhow!("could not find function \"{}\"", method))?
-        .typed::<(), ()>(&mut store)?;
-
-    metadata.call(&mut store, ())?;
-    Ok(store.data().msg.clone())
+        .exports
+        .get_function(method)
+        .with_context(|| format!("could not find function \"{}\"", method))?
```
Comment (shamilsan) [inline gcli/src/meta/executor.rs:70]:
```suggestion
        .with_context(|| format!("could not find function \"{method}\""))?
```

File: gcli/src/meta/executor.rs [hunk 6]
```diff
@@ -179,116 +173,116 @@ mod env {
...
 
-                memory
-                    .clone()
-                    .write(caller.as_context_mut(), err, &len.to_le_bytes())
-                    .map_err(|e| {
-                        log::error!("{:?}", e);
-                        Trap::new(TrapCode::MemoryOutOfBounds.trap_message())
-                    })?;
+                memory.write(err, &len.to_le_bytes()).map_err(|e| {
+                    log::error!("{:?}", e);
```
Comment (shamilsan) [inline gcli/src/meta/executor.rs:222]: Mixed style

```suggestion
                    log::error!("{e:?}");
```

**Review by shamilsan — APPROVED** [review]
Good job 👍

File: Cargo.toml (hunk 2 above)
Comment (ark0f) [reply Cargo.toml:306]: We are going to update wasmi so there will be only one version

**Pull request merged**